vmdb/app/views/layouts/quadicon/_ext_management_system.html.haml

```diff
@@ -1,4 +1,12 @@
-- if @settings[:quadicons][@layout == 'ems_infra' ? :ems : :ems_cloud]
+- db = case @layout
+- when "ems_infra"
+  - :ems
+- when "ems_cloud"
+  - :ems_cloud
+- else
+  - :ems_container
```

Comment:
Well if you create a method to calculate the icon. Maybe a helper method? I would look much better than programming in-line in the HAML template.